### DISCUSSION TITLE
Startup sequence zigbee2mqtt

### DISCUSSION THREAD

**Discussion by timoline** [Koenkk/zigbee2mqtt#5653] [General]
Last time I checked the startup sequence of zigbee2mqtt is
- start serial connection
- start mqtt connection

I think it's better to 
- start mqtt connection
- start serial connection

This way you can log the status of the serial connection to the mqtt broker
Also if for some reason the serial connection suddenly failes, you can publish this to the mqttbroker

**Comment by timoline** [comment]
And maybe:

- start frontend
- start mqtt connection
- start serial connection

**Reply by sjorge** [reply]
The frontend will not work  if mqtt is operation, you can see this when your broker is down at startup.

**Reply by timoline** [reply]
currently not, but technically I think it should be possible to change this.....
